examples/example.cpp

```diff
@@ -4,43 +4,108 @@
 // Distributed under the Boost Software License, Version 1.0. (See accompanying
 // file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 //
+//[ main
+#include <boost/system/error_code.hpp>
 
-#include <nudb/nudb.hpp>
-#include <cstddef>
-#include <cstdint>
+#include <nudb/xxhasher.hpp>    // xxhasher
+#include <nudb/create.hpp>      // create
+#include <nudb/store.hpp>       // store
+#include <nudb/file.hpp>        // path_type
+
+#include <iostream>
+#include <cstdint>              // std::uint32_t, std::uint64_t
+
+// NuDB functions may return error_code defined by
+bool is_success(const boost::system::error_code & ec){
+    return
+        // boost system library
+        ec == boost::system::errc::success
+        // or nudb error header
+        || ec == nudb::error::success;
```

Comment:
This is kind of interesting. "success should not be part of nudb::error".  That is a little unusual, but not really a problem.  I'm aware that error_code is implicitly convertible to bool according to some rules.  We can explain this rather than making the is_success.  This will make the main example code and explanation significantly more elegant.  (FYI I think using implicit conversions in API's in this way is a bad idea - but we'll make it work and it will be fine.)